### PR TITLE
Improve search suggestions android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/31607

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java [hunk 1]
```diff
@@ -235,7 +224,7 @@ public void onDestroy() {
     public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
         super.onCreatePreferences(savedInstanceState, rootKey);
 
-        mManagedPreferenceDelegate = createManagedPreferenceDelegate();
+        // mManagedPreferenceDelegate = createManagedPreferenceDelegate();
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java:227]: I suppose this was meant to be removed?

File: android/java/org/chromium/chrome/browser/settings/BraveSearchEnginesPreferences.java [hunk 2]
```diff
@@ -32,16 +45,76 @@ public void onResume() {
...
+
+        mSearchSuggestions = (ChromeSwitchPreference) findPreference(PREF_SEARCH_SUGGESTIONS);
+        mSearchSuggestions.setOnPreferenceChangeListener(this);
+        mSearchSuggestions.setManagedPreferenceDelegate(mManagedPreferenceDelegate);
+
+        mShowAutocompleteInAddressBar =
+                (ChromeSwitchPreference) findPreference(PREF_SHOW_AUTOCOMPLETE_IN_ADDRESS_BAR);
+        mShowAutocompleteInAddressBar.setOnPreferenceChangeListener(this);
+
+        mAutocompleteTopSites =
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/settings/BraveSearchEnginesPreferences.java:87]: I'm not sure why `AutocompleteInAddressBar` and `AutocompleteTopSites`  are in `Search engine preference`. They are technically have nothing to do with search engine. Was this move agreed with design team?

File: android/java/res/xml/brave_privacy_preferences.xml [hunk 3]
```diff
@@ -168,7 +168,7 @@
             android:order="25"
             android:title="@string/sns_title" />
 
-        <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        <!-- <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
```
Comment (samartnik) [inline android/java/res/xml/brave_privacy_preferences.xml:171]: If we moved these to other place, this should be just deleted.

File: android/java/org/chromium/chrome/browser/settings/BraveSearchEnginesPreferences.java (hunk 2 above)
Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/settings/BraveSearchEnginesPreferences.java:87]: yes it was agreed upon. we will be following up with more granular details for the option. CC @timchilds @anthonypkeane

File: android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java (hunk 1 above)
Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java:227]: sorry it's my bad.

File: android/java/res/xml/brave_privacy_preferences.xml (hunk 3 above)
Comment (deeppandya) [reply android/java/res/xml/brave_privacy_preferences.xml:171]: sorry it's my bad. updating now.

File: browser/ui/android/strings/android_brave_strings.grd [hunk 4]
```diff
@@ -3967,22 +3967,23 @@ If you don't accept this request, VPN will not reconnect and your internet conne
...
+    <message name="IDS_KILL_SWITCH_TUTORIAL_TEXT_3" desc="Vpn kill switch tutorial text">
       3. Toggle <ph name="ALWAYS_ON_TUTORIAL_START">&lt;always_on_tutorial&gt;</ph>Always-on VPN<ph name="ALWAYS_ON_TUTORIAL_END">&lt;/always_on_tutorial&gt;</ph> to enable it.
     </message>
-    <message name="IDS_KILL_SWITCH_ACTION_TEXT" desc="Vpn kill swicth tutorial action text">
+    <message name="IDS_KILL_SWITCH_ACTION_TEXT" desc="Vpn kill switch tutorial action text">
       Open Android settings
     </message>
-    <message name="IDS_IN_APP_UPDATE_TEXT" desc="Text for in app update snackbar">
+    <message name="IDS_IN_APP_UPDATE_TEXT" desc="Text for in git diff
+      app update snackbar">
```
Comment (mkarolin) [inline browser/ui/android/strings/android_brave_strings.grd:3986]: Looks like an accidental paste of `git diff` in here.

Comment (deeppandya) [reply browser/ui/android/strings/android_brave_strings.grd:3986]: oops. good catch

**Review by samartnik — APPROVED** [review]
lgtm

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by tapanmodh — APPROVED** [review]
lgtm

**Pull request merged**